lib/src/test/kotlin/starknet/provider/handlers/BasicHttpErrorHandlerTest.kt

```diff
@@ -1,32 +1,88 @@
-import com.swmansion.starknet.service.http.handlers.BasicHttpErrorHandler
+import com.swmansion.starknet.provider.exceptions.RpcRequestFailedException
+import com.swmansion.starknet.provider.gateway.GatewayProvider
+import com.swmansion.starknet.provider.rpc.JsonRpcProvider
+import com.swmansion.starknet.service.http.HttpResponse
+import com.swmansion.starknet.service.http.HttpService
 import org.junit.jupiter.api.Assertions.*
 import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
 
-class BasicHttpErrorHandlerTest {
+class HttpErrorHandlerTest {
     @Test
-    fun `handler falls back to basic exception on unknown format`() {
-        val handler = BasicHttpErrorHandler()
+    fun `gateway handler falls back to basic exception on unknown format`() {
         val message = "{\"status_code\": 500, \"status_message\": \"error\"}"
+        val httpServiceMock = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(false, 500, message)
+        }
+        val provider = GatewayProvider("", "", StarknetChainId.TESTNET, httpServiceMock)
+        val request = provider.getTransaction(Felt(1))
 
         val exception = assertThrows(RequestFailedException::class.java) {
-            handler.handle(message)
+            request.send()
         }
         assertFalse(exception is GatewayRequestFailedException)
         assertEquals(message, exception.message)
     }
 
     @Test
-    fun `handler parses gateway error`() {
-        val handler = BasicHttpErrorHandler()
+    fun `rpc handler falls back to basic exception on unknown format`() {
+        val message = "{\"status_code\": 500, \"status_message\": \"error\"}"
+        val httpServiceMock = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(false, 500, message)
+        }
+        val provider = JsonRpcProvider("", StarknetChainId.TESTNET, httpServiceMock)
+        val request = provider.getTransaction(Felt(1))
+
+        val exception = assertThrows(RequestFailedException::class.java) {
+            request.send()
+        }
+        assertFalse(exception is RpcRequestFailedException)
+        assertEquals(message, exception.message)
+    }
+
+    @Test
+    fun `gateway handler parses gateway error`() {
         val message = "{\"status_code\": 500, \"message\": \"error\"}"
+        val httpServiceMock = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(false, 500, message)
+        }
+        val provider = GatewayProvider("", "", StarknetChainId.TESTNET, httpServiceMock)
+        val request = provider.getTransaction(Felt(1))
 
         val exception = assertThrows(GatewayRequestFailedException::class.java) {
-            handler.handle(message)
+            request.send()
         }
         assertEquals("error", exception.message)
     }
+
+    @Test
+    fun `rpc handler parses rpc error`() {
+        val message = "{\n" +
```

Comment:
Please use https://realkotlin.com/tutorials/2018-06-26-multiline-string-literals-in-kotlin/